scripts/build_common.sh

```diff
@@ -76,14 +76,19 @@ configure_boot_method ()
   # Change PetaLinux project config to change the boot method
   echo -e "\nModifying project config for ${BOOT_METHOD} boot support...\n"
 
-  bash ./config.boot_method.${BOOT_METHOD}.sh ${PETALINUX_BOARD_NAME} ${PETALINUX_BOARD_FAMILY} ${INITRAMFS_IMAGE}
+  bash ${PETALINUX_CONFIGS_FOLDER}/project/config.boot_method.${BOOT_METHOD}.sh ${PETALINUX_BOARD_NAME} ${PETALINUX_BOARD_FAMILY} ${INITRAMFS_IMAGE}
 
   petalinux-config --silentconfig
 
 }
 
 build_bsp ()
 {
+  BOOT_METHOD=$1
+  BOOT_SUFFIX=$2
+  INITRAMFS_IMAGE=$3
```

Comment:
The shell variable ${INITRAMFS_IMAGE} is used in configure_boot_method, but not on the 'petalinux-build -c' command.  The 'petalinux-build -c' command always specifies  ${PETALINUX_BUILD_IMAGE} ('petalinux-build -c  ${PETALINUX_BUILD_IMAGE}'), which is always set to 'avnet-image-full', so it appears that PetaLinux never builds the "_MINIMAL" image with the 'avnet-image-minimal' rootfs config.

The more I look at this the more I think we can pare down 'build_bsp' down to two command arguments for $BOOT_METHOD <INITRD | EXT4> and $BOOT_SUFFIX <MINIMAL | FULL> and change $INITRAMFS_IMAGE and $ROOTFS_CONFIG.  Add some logic to the script that if $BOOT_SUFFIX is set to 'FULL' then $ROOTFS_CONFIG gets set to 'avnet-image-full' else if 'MINIMAL' then 'avnet-image-minimal'.

We could simplify this even further down to one 'build_bsp' command argument for INITRD or EXT4.  In this scheme the INITRD will set the boot suffix to MINIMAL and *ALWAYS* build the 'avnet-image-minimal rootfs, and likewise the EXT4 option will set the boot suffix to FULL and *ALWAYS* build the 'avnet-image-full' rootfs.  This will take some extra thought to make sure there isn't a corner case where a 'avnet-image-full' rootfs isn't needed for an INITRD OS image.  Your thoughts?

Reply:
Even if Today we are always using the 'avnet-image-full' image for EXT4 (or 'avnet-image-minimal'/'avnet-image-full' images for INITRD), I prefer not to hardcode this in the build_common.sh.

The reason is that some customers may want to create their own image (based on the Avnet images), so I think it's easier if all the important variables they may want to change are explicitly defined in the board scripts.

(For example, sometimes when I am developing/testing something for a board/project, I create a new image, and just change the variable in the board script)

Reply:
I've reverted these so they will continue to be explicit